README.md

```diff
@@ -191,14 +192,26 @@ const options = {
 };
 ```
 
+### Using callbacks for `from`
+You can also specify a callback that returns a string or a regular expression. The callback receives the name of the file in which the replacement is being performed, thereby allowing the user to tailor the search string. The following example uses a callback to produce a search string dependent on the filename:
+
+```js
+const options = {
+  files: 'path/to/file',
+  from: (file) => RegExp(`${foo('SomePattern[A-Za-z-]+', file)}`, g);,
+  to: 'bar',
+};
+```
+
 ### Using callbacks for `to`
-As the `to` parameter is passed straight to the native [String replace method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace), you can also specify a callback. The following example uses a callback to convert matching strings to lowercase:
+As the `to` parameter is passed to the native [String replace method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace), you can also specify a callback. This callback provides as an extra argument the name of the file in which the replacement is being performed. The following example uses a callback to convert matching strings to lowercase and appends the file extension to it:
 
 ```js
+var path = require('path');
```

Comment:
Sorry missed this, could you change `var` to `const` please, as the project is using ES6 throughout.

Reply:
Actually maybe remove the `path` reference altogether, as I'd like to keep the examples as simple as possible, without external dependencies.